packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,266 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP connection and channel."""
+
+import asyncio
+import logging
+import threading
+import time
+import json
+import yaml
+from asyncio import CancelledError
+from threading import Thread
+from typing import Any, Dict, List, Optional, Set, cast
+
+from fetch.p2p.api.http_calls import HTTPCalls
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.helpers.base import locate
+from aea.mail.base import AEAConnectionError, Address, Envelope, EnvelopeContext
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPChannel:
+    """A wrapper for an RESTful API."""
+
+    def __init__(
+        self,
+        address: Address,
+        api_spec: str,
+        provider_addr: str,
+        provider_port: int,
+        excluded_protocols: Optional[Set[PublicId]] = None,
+    ):
+        """
+        Initialize a channel.
+
+        :param address: the address
+        """
+        self.address = address
```

Comment:
note, in the P2PChannel, the P2PChannel acts as the client. Here, the HTTPChannel acts as a server.